jupyterhub/spawner.py

```diff
@@ -27,63 +27,102 @@
 from .traitlets import Command, MemorySpecification
 from .utils import random_port
 
+
 class Spawner(LoggingConfigurable):
-    """Base class for spawning single-user notebook servers.
-    
+    """
+    Abstract base class for spawning single-user notebook servers.
+
     Subclass this, and override the following methods:
-    
+
     - load_state
     - get_state
     - start
     - stop
     - poll
+
+    An instance of a `Spawner` subclass is created for each user.
     """
-    
+
     db = Any()
     user = Any()
     hub = Any()
     authenticator = Any()
     api_token = Unicode()
-    ip = Unicode('127.0.0.1',
-        help="The IP address (or hostname) the single-user server should listen on"
+
+    ip = Unicode(
+        '127.0.0.1',
+        help="""
+        The IP address (or hostname) the single-user server should listen on.
+
+        The JupyterHub proxy implementation should be able to send packets to this interface.
```

Comment:
Hmm, there's only one proxy running in our case, and I also want to explicitly specify which one it is (the one fronting JupyterHub) since there can be many number of them between the user and the instance (on wikimedia's install there's at least 3!).